### ISSUE TITLE
Version 0.2.0 fails with dry-validation.

### ISSUE THREAD

**Issue by michalwarda** [dry-rb/dry-configurable#12]
When using `dry-validation` `0.9.5` with `dry-configurable` `0.2.0` I get an error.
It happens when trying to require `dry-validation`.

I don't think I can provide any more inside but I can try to explain more if you want to.

`***` is a path to spec file.
Here is the stacktrace:
```
bundler: failed to load command: rspec (/Users/x/.rbenv/versions/2.3.0/bin/rspec)
ArgumentError: unknown keywords: sanitizer, json, form
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-configurable-0.2.0/lib/dry/configurable.rb:83:in `setting'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation/schema/class_interface.rb:25:in `<class:Schema>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation/schema/class_interface.rb:6:in `<module:Validation>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation/schema/class_interface.rb:5:in `<module:Dry>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation/schema/class_interface.rb:4:in `<top (required)>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation/schema.rb:17:in `require'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation/schema.rb:17:in `<top (required)>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation.rb:5:in `require'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry/validation.rb:5:in `<top (required)>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry-validation.rb:1:in `require'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.9.5/lib/dry-validation.rb:1:in `<top (required)>'
 ***_spec.rb:2:in `require'
  ***_spec.rb:2:in `<top (required)>'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/configuration.rb:1435:in `load'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/configuration.rb:1435:in `block in load_spec_files'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/configuration.rb:1433:in `each'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/configuration.rb:1433:in `load_spec_files'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/runner.rb:100:in `setup'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/runner.rb:86:in `run'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/runner.rb:71:in `run'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/runner.rb:45:in `invoke'
  /Users/x/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/exe/rspec:4:in `<top (required)>'
  /Users/x/.rbenv/versions/2.3.0/bin/rspec:22:in `load'
  /Users/x/.rbenv/versions/2.3.0/bin/rspec:22:in `<top (required)>'
```

**Comment by AMHOL** [comment]
Strange, seems like when you pass a hash as the default Ruby treats it as keyword arguments

**Comment by AMHOL** [comment]
Had to yank 0.2.0 for now, possible bug in mri

```ruby
class Foo
  def self.options(value = nil, **options)
    puts value.inspect
    puts options.inspect
  end

  def self.kwarg(value = nil, kw: nil)
    puts value.inspect
    puts kw.inspect
  end

  def self.splat(*args, kw: nil)
    puts args.inspect
    puts kw.inspect
  end
end

Foo.options({})
# nil
# {}
Foo.kwarg({})
# nil
# nil
Foo.splat({})
# []
# nil
```

**Comment by AMHOL** [comment]
https://bugs.ruby-lang.org/issues/12717

**Comment by Kukunin** [comment]
Are you going to workaround this is somehow?
It prevents us from updating, and ruby bug would take soo long to resolve that bug.

**Comment by AMHOL** [comment]
@Kukunin I yanked `0.2.0` so the latest released version is back to `0.1.7` (see https://rubygems.org/gems/dry-configurable), is it still trying to install `0.2.0` for you?
